ipydeps/__init__.py

```diff
@@ -428,19 +428,21 @@ def pip(pkg_name, verbose=False, progress=False):
         pip_exec = _get_pip_exec(_config_options)
         package_errs = {}
```

Comment:
@gershwinlabs After a couple of adjustments, I like your idea of bringing the package iteration and progress bar out to pip() (much cleaner!) -- as for wrapping it up in a function for added cleanliness, do you mean lines 429-436 in this commit?